pkg/engine/deployment.go

```diff
@@ -196,6 +196,10 @@ func newDeployment(ctx *Context, info *deploymentContext, opts *deploymentOption
 	}()
 
 	opts.trustDependencies = proj.TrustResourceDependencies()
+	if !opts.trustDependencies && opts.ContinueOnError {
```

Comment:
oh we should just clean this up, TrustResourceDependencies is _always_ true. If you raise another PR to just remove the trustDependencies methods and fields we can approve that quickly and then not worry about it here.

Reply:
Yeah I noticed it's always true currently, but got stopped by the comment about it potentially being useful for bringing up new languages.  But I guess really new languages should just get this right and we shouldn't have to worry about it in the engine.

https://github.com/pulumi/pulumi/pull/15745